card/lib/generators/card/USAGE

```diff
@@ -0,0 +1,21 @@
+Description
+
+Examples
+  wagn generate card new cards pre_fix
```

Comment:
This is really the wrong place for this discussion.  I think we need to talk so we are sure we are really on the same page.

What frustrates me is the only reason there is a decko-rails is that we are not ready to change the branding and go away from wagn.  Lets imaging a future in which there is also a decko-sinatra or decko-pick_some_framework, and in that context, generating an app deck (what wagn new does now) would be just one of the frameworks it generates for.  So, decko new sinatra or decko new my_framework would implicate decko-sinatra or decko-my_framework.  Maybe plain 'decko new' implied decko-rails by default, or maybe it is included in decko.

Any way you slice that, it is wrong to move the basic engine support from decko-rails back to wagn (or decko).  The decko-rails gem would need to have all the engine stuff.  Engines are a rails concept, so you wouldn't have it for decko-sinatra, right?  I guess that things like assets are ideally shared, with adapting frameworks going into deck-platform gems, but the java/coffeescript, images and such should be in a shared gem (wagn/decko).  On the other hand, I think it is ok to combine decko and decko-rails and have the other glue modules adapt the shared parts from that combined gem.

To me, decko-rails is a shim we need to not have to re-brand all at once.  To you it is the support for this use case that you don't want to get in the way of "normal" wagn functioning.  I'm supportive of the idea of not interfering with the common (now only) use case.  I'm even supportive of the idea of making the usage of this new case (engine decks) smoother from a UI standpoint, but it is pre-mature to optimize for that now.

Note that I agreed with your point about not generating a migration in 'rails g card', that file should be included in the card gem.  Did you get my point about using that file in loading the schema in the standard flow?  Rather than compressing all the migrations up to a point into a schema.rb file, they get compressed into a single migration that takes the db structure from empty to the current structure.  It still needs to be set apart from the rest of the migrations as it may take you past some of the other migrations still needed by decks in the field.

I'm trying to get feedback on the CLI, and I do appreciate that.  I'm also trying to improve the generators as they are.  We should not need the wagn command to access the card generators.  Those are rails things and should extend the rails framework.  I'm more concerned about this sort of question about where they belong code-wise.  That's why I'm objecting so strongly to your desire to move code back into wagn.

Note that given your stated goals here, it is actually wrong for wagn command to build off of the rails command stuff.  Well, it's fine for wagn, but not what we would want for decko.  Maybe for decko-rails if it implemented a command (decko most naturally and a conflict).

Yes, we want to make the commands as natural as possible, but no, we don't care that much about the details of what command you run for which case.  We can create wrapper scripts for that when we know what the use cases are.  The whole reason I created these PRs way before they were ready is so I could follow your feedback as I went along.  Now, I'm not feeling particularly acknowledged for the work done.

I'm not that worried about getting an engine to work in a simple application started from scratch.  I'm concerned about my use case of an enormous rails app that I wan't to add cards to.  If I try to present this as a solution to my colleagues when the basic plumbing isn't right, I'll be sunk from the start.  They won't care if the CLI is natural, but about whether the steps work and don't break things.

I'm working on the basic functioning of these work flows because the first accomplishment I want is to add cards to an empty rails app and basically get a functioning wagn.  Then I'll be ready to try to add cards to an existing app.  I'll want namespaces before I can add it to our complex app.